### ISSUE TITLE
Propose New Layer Type (Control) for Handling Forward and Backward (Gradient Flow) with Control Mask over samples, Supporting Backend Agnosticism

### ISSUE THREAD

**Issue by ikhwanperwira** [keras-team/keras#20360]
I want to propose a new layer type (`control`) that sits on `keras/src/layers/control/`. It's consisting of `SkipGrad`, `StopGrad`, and `Switch`:

* **StopGrad**: It's not only a wrapper version of  `ops.stop_gradient()` but also supports a controlled version for certain samples, whether a gradient for i-sample is allowed to propagate or blocked (e.g., due to a lack of target for certain task in the context of multi-task learning).
* **Switch**: Layer to control which layer to forward pass between two layers (both have the same shape). The gradient flowed to the switched layer but not the unswitched one. (e.g., using real sample if control mask is 1, otherwise using fake sample in the context of GAN)
* **SkipGrad**: Layer to control whether a layer should be skipped during backward propagation. Unlike the stop gradient, where it affects the previous layers (gradient also stopped), the `SkipGrad` is still allowing the previous layers to flow.

Those three controls can be understand intuitively using this truth table:
![image](https://github.com/user-attachments/assets/674d51df-fcbb-4187-9c6b-35dff65c538a)

For the proof of concept, I actually already defined them when training static GAN, where the training stages is controlled by control mask. I already define `SkipGrad`, `StopGrad`, and `Switch`, the three has been used in the case  of GAN.
![Untitled](https://github.com/user-attachments/assets/b1631fdd-a1d2-420d-ba55-230851f91052)

Here is the result of how I trained it locally until 100 epochs:
![gan_image_evolution_2(1)](https://github.com/user-attachments/assets/99fae2de-21f2-498b-9c00-52aa70f1b3f4)

For another use case, it's possible that these control layers are useful in reinforcement learning or multi-task learning, specifically domain-specific learning. In fact, I was thinking about this idea based on my main work, where I'm working on ECG signal data. There is an abnormal case where it is supposed to be more specific to handle it due to lack of target while still allowing normal case for training.

Indeed I can use manual gradient control, but I think it's higher-level to control the gradient flow through control mask than overriding `train_step()`. Moreover, I don't see any docs in keras.io that talking about backend agnosticism to control gradient flow. A closer docs I can get is `ops.custom_gradient`. Perhaps, it can be useful for another control layers in future. Some literatures like SkipNet or ResNet are using control mechanism about how forward propagation and backward propagation should be handled. But we need more generalization of how the gradient should be flow.

Here is the Google Colaboratory of my POC above about [static_dcgan](https://colab.research.google.com/drive/1aUhoyvzhD9IA0gATBZ7FbpaTyauL4SxL?usp=sharing])

So, what do you think? Let me know if I'm allowed to make the draft pull request.

**Comment by ikhwanperwira** [comment]
Thank you for your feedback regarding my proposal. I understand the team’s perspective on not including these layers in the core API, and I appreciate the suggestion to contribute an example instead.

I believe that creating a detailed example that following guideline will take some time, as I want to ensure it effectively showcases the potential applications of the SkipGrad, StopGrad, and Switch layers. I would like to request a bit more time to prepare this example thoroughly.
